docs-client/build.gradle

```diff
@@ -28,9 +30,21 @@ task buildWeb(type: NpmTask) {
 
     args = ['run', 'build']
 
-    inputs.dir('src')
-    inputs.files('package.json', 'package-lock.json')
+    // note that node_modules isn't cached due to potential cache misses per architecture
+    inputs.property("node.version", node.version)
+    // build per OS type to preserve behavior of packaging jar from linux
+    inputs.property("os.familyName", current().getFamilyName())
+    inputs.dir('src').withPathSensitivity(PathSensitivity.RELATIVE)
+            .withPropertyName("sources files")
+    inputs.files('package.json', 'package-lock.json', 'tsconfig.json', 'tsconfig-webpack.json',
+                 'webpack.config.ts')
+            .withPathSensitivity(PathSensitivity.RELATIVE)
+            .withPropertyName("npm configuration files")
+
     outputs.dir('build/web')
+
+    // TODO: consider disabling cacheability once arm actions runners are available
+    outputs.cacheIf { true }
```

Comment:
Question: The PR description says "Added a cacheIf condition only if the build machine is linux". Is `cacheIf { true }` an intentional behavior?

Reply:
Oops, I changed the direction once and forgot to update the description 😅  Thanks for pointing it out 👍

Reply:
Let me go ahead and merge this so that I can collect some feedback before our next meeting 😄